src/lib/TwaManifest.ts

```diff
@@ -222,16 +207,39 @@ export class TwaManifest {
 
     const fullStartUrl: URL = new URL(webManifest['start_url'] || '/', webManifestUrl);
 
-    const shortcuts: ShortcutInfo[] = (webManifest.shortcuts || [])
-        .map((s: WebManifestShortcutJson) => new ShortcutInfo(s, webManifestUrl))
-        .filter((s: ShortcutInfo) => s.isValid())
-        .filter((_: ShortcutInfo, i: number) => i < 4);
+    const shortcuts: ShortcutInfo[] = [];
+
+    if (webManifest.shortcuts) {
```

Comment:
nit: `for (const s of (webManifest.shortcuts || [])) {`
to reduce indentation.

I'm not too familiar with TS but it seems to me that webManifest.shortcuts is guaranteed to be a list instead of a null or undefined though?

Reply:
This is a WebManifest shortcut, defined here: https://github.com/GoogleChromeLabs/llama-pack/blob/master/src/lib/types/WebManifest.ts#L40

It can be undefined.

Reply:
changed the for loop.